records/2022-11-22.security.blazer.md

```diff
@@ -0,0 +1,45 @@
+# Introducing Blazer and its Implementation for Notify
+
+Date: 2022-11-22
+
+## Status
+
+**DRAFT**
+
+## Context
+In order to have a thorough understanding of data in our system, Notify devs wanted a collaborative tool which would allow them to explore the database. The tool would allow us to store queries and share them between developers. An added bonus would be if the tool graphed results in a meaningful manner. The goal of this ADR is to explain the implementation of the tool that was chosen for exploring the database.
+
+## Blazer
+
+Going through a list of products that are available and meet our main requirement of collaboration and graphing we decided on https://github.com/ankane/blazer
+
+### Implementation
+
+Introducing blazer into our system requires us to do the following:
+
+1. Blazer needs access to the main Notify DB
+2. Blazer needs to be secure and auditable
+
+## Deployment
+
+### Blazer Container
+The first step is to add Blazer to the Notify ECR. This way Notify has control over the image that is deployed to our production system. The downside of this is that Notify devs need to maintain this image by pushing security updates
```

Comment:
Do we have any mechanism in place to report if a security patch is required by us?

Reply:
Yes, Renovate is active on the repo for patching and GitHub Security will also raise vulnerability alerts.  There is currently one open that does not impact us and can be dismissed:
https://github.com/cds-snc/notification-lambdas/security/dependabot/1

Reply:
added this under the Blazer Container section